src/pages/contact.js

```diff
@@ -83,37 +83,31 @@ function Contact(takwimu) {
     count += 1;
   }
 
-  const changeActiveContent = useCallback(
-    index => {
-      setCurrentSectionIndex(index);
-      const activeElement = document.getElementById(
-        contentHeadings[index].link
-      );
-      window.scrollTo(0, activeElement.offsetTop - 90);
-    },
-    [contentHeadings]
-  );
-
+  const [current, setCurrent] = useState(-1);
   useEffect(() => {
-    const currentIndex = contentHeadings.findIndex(
-      x => x.link === window.location.hash.replace('#', '')
-    );
-    if (currentIndex !== -1) {
-      changeActiveContent(currentIndex);
+    if (current === -1) {
+      return;
     }
-  }, [changeActiveContent, contentHeadings]);
+    const { link } = contentHeadings[current];
+    const sectionEl = document.getElementById(link);
+    if (sectionEl) {
+      window.scrollTo(0, sectionEl.offsetTop - 90);
+    }
+  }, [contentHeadings, current]);
 
   const router = useRouter();
   useEffect(() => {
-    router.events.on('hashChangeComplete', () => {
-      const index = contentHeadings.findIndex(
-        heading => heading.link === window.location.hash.slice(1)
+    const handleHash = () => {
+      setCurrent(
+        contentHeadings.findIndex(
+          heading => heading.link === window.location.hash.slice(1)
+        )
       );
-      if (index !== -1) {
-        changeActiveContent(index);
-      }
-    });
-  }, [changeActiveContent, contentHeadings, router]);
+    };
+
+    handleHash();
+    router.events.on('hashChangeComplete', handleHash);
```

Comment:
Shouldn't we `Router.events.off` in a `return` as well?

Reply:
Good one